src/store/lists/actions.js

```diff
@@ -291,43 +301,69 @@ export default {
         const newTagObj = responseList.tags.find(tag => tag.title === tagTitle);
+        let { tags } = itemObj;
+        
+        tags = [...tags, newTagObj.id];
+
+        if (rootGetters['items/edittingItemObj']) {
+          commitFromRoot('updateItemFieldLocally', {
+            field: 'tags',
+            value: tags,
+          });
+        }
 
+        dispatchFromRoot('items/_saveItemOnServer', { ...itemObj, tags });
         commit('setCurrentListObj', responseList);
-        commitFromRoot('updateItemFieldLocally', {
-          field: 'tags',
-          value: [...rootGetters['items/edittingItemObj'].tags, newTagObj.id],
-        });
-        dispatchFromRoot('items/_saveItemOnServer');
       })
       .catch(error => {
         notifyAboutError(error);
-        dispatchFromRoot('items/_fetchItemById', {
-          id: rootGetters['items/edittingItemObj'],
-          cancelToken: null,
-        });
+        dispatch('_fetchListById', { id: listObj.id, cancelToken: null });
+      })
+      .finally(() => {
+        commitFromRoot('decreaseExplicitRequestsNumber');
       });
   },
 
-  _addCategoryForListAndItem({ dispatch, getters, rootGetters }, { listObj, categoryTitle }) {
-    return dispatch('_updateList', listObj)
-      .then(() => {
-        const newCategoryObj = getters.currentListCategories?.find(
+  _addCategoryForListAndItem(
+    { commit, rootGetters, dispatch },
+    { listObj, itemObj, categoryTitle },
+  ) {
+    commitFromRoot('increaseExplicitRequestsNumber');
+
+    return this.$config.axios
+      .patch(
+        `${this.$config.apiBasePath}list/update/${listObj.id}`,
+        {
+          title: listObj.title,
+          isPrivate: listObj.isPrivate,
+          tags: listObj.tags,
+          categories: listObj.categories,
+        },
+      )
+      .then(({ data: responseList }) => {
+        const newCategoryObj = responseList.categories?.find(
           category => category.title === categoryTitle,
         );
+        let { category } = itemObj;
 
-        commitFromRoot('updateItemFieldLocally', {
-          field: 'category',
-          value: newCategoryObj.id,
-        });
+        category = newCategoryObj.id;
 
-        dispatchFromRoot('items/_saveItemOnServer');
+        if (rootGetters['items/edittingItemObj']) {
+          commitFromRoot('updateItemFieldLocally', {
+            field: 'category',
+            value: category,
+          });
+        }
+
+        dispatchFromRoot('items/_saveItemOnServer', { ...itemObj, category });
+        commit('setCurrentListObj', responseList);
       })
       .catch(error => {
         notifyAboutError(error);
-        dispatchFromRoot('items/_fetchItemById', {
-          id: rootGetters['items/edittingItemObj'],
-          cancelToken: null,
-        });
+        dispatch('_fetchListById', { id: listObj.id, cancelToken: null });
+      })
+      .finally(() => {
+        commitFromRoot('decreaseExplicitRequestsNumber');
       });
```

Comment:
looks like tags and categories looks pretty the same 
let's think how we can make it universal if we could and should we or not